### ISSUE TITLE
Issues with installing on Ubuntu 20.04 focal on Hyper-V

### ISSUE THREAD

**Issue by Denys-Priadko** [ComPlat/chemotion_ELN#377]
Hello, Chemotion team!

We have read your paper and are eager to see Chemotion in action!

Unfortunatelly, we could not as yet install it. We tried several approaches: 
1. Using docker - development branch
2. Using the script - development branch. 
3. Using docker - master branch.

**Approach 1. Using docker - development branch:** 

The docker image was pulled. On starting the service to initialize the database container, it notifies us we need to install missing gems and stops. 
So using 
#docker-compose run app bash 
app_container# bundle install
app_container# bundle exec rake db:create
app_container# bundle exec rake db:migrate
In config/datacollectors.yml.example, a “   ‘   “ character seems to be missing at the end of line 37 (the last line on screenshot below), which causes errors with db:seed step or importing common_templates. 
![image](https://user-images.githubusercontent.com/59832692/109198138-22f5bc00-779e-11eb-80cf-85d6236c5db2.png)

Adding the “   ‘   “ character. 
	app_container# bundle exec rake db:seed
	app_container# rake ketcherails:import:common_templates
Until now everything seems to work, except it keeps saying required Ruby is not installed. The version installed in the pulled container differs very slightly, and doing ‘# rvm install “ruby-2.5.6” did not help in the past, so this time I’ll leave ruby as is. 

On the precompiling step, there is an error: 
![image](https://user-images.githubusercontent.com/59832692/109198173-31dc6e80-779e-11eb-804a-5118b387bac3.png)

Installing browserify:
app_container#  npm i browserify
![image](https://user-images.githubusercontent.com/59832692/109198229-43257b00-779e-11eb-80fe-1b5f1fdaf007.png)

Next attempt at precompiling also returns an error: this time it looks for rfb.js not where it is installed. 
![image](https://user-images.githubusercontent.com/59832692/109198249-4b7db600-779e-11eb-89f0-da44ac3ac442.png)

Correcting the path in /app/assets/javascripts/libCnC/CnC.js:

From 
![image](https://user-images.githubusercontent.com/59832692/109198285-546e8780-779e-11eb-95e4-5c575cda4f6e.png)
To 
![image](https://user-images.githubusercontent.com/59832692/109198299-59333b80-779e-11eb-9241-81d9b12c346e.png)

Now react-svg-file-zoom-pan-latest is declared to be missing: 

![image](https://user-images.githubusercontent.com/59832692/109198315-5df7ef80-779e-11eb-937f-2f9e6bd50eb3.png)

Installing it: 
app_container# npm i react-svg-file-zoom-pan-latest
![image](https://user-images.githubusercontent.com/59832692/109198341-66502a80-779e-11eb-9fcc-89ebad446d53.png)

Trying precompile step again
![image](https://user-images.githubusercontent.com/59832692/109198366-6f40fc00-779e-11eb-938b-f21a23b71bda.png)

Reading here, it seems like the problem might be the new babel that comes with the new browserify.
Trying 
app_container#npm i browserify@~8.0.0 
![image](https://user-images.githubusercontent.com/59832692/109198389-7700a080-779e-11eb-9dfc-9ffbdd4e063c.png)

Then precompile again. Another error: 

Trying npx browserlist@latest --update-db, as advised in the output, doesn’t help. 

If I skip the precompile step, the same errors come later when the system is up and I’m trying to connect to the server via browser: 

![image](https://user-images.githubusercontent.com/59832692/109198428-7ff17200-779e-11eb-8d7f-72572ad99809.png)

To make that work, I modified the run.sh to include the steps we’ve just gone through:
![image](https://user-images.githubusercontent.com/59832692/109198469-8aac0700-779e-11eb-8f0b-97591b4f6128.png)

As seen from the browser: 
![image](https://user-images.githubusercontent.com/59832692/109198498-94ce0580-779e-11eb-99bf-6f9a244507a0.png)

**2. Approach 2 - using the script (dev. branch)**

After linking 
#sudo ln -s /bin/mkdir /usr/bin/mkdir
to avoid the error while nokogiri installation, at the end we get: 

![image](https://user-images.githubusercontent.com/59832692/109198522-9b5c7d00-779e-11eb-8f31-a78698c2837a.png)

**3. Approach 3 - docker with master branch**

When, analogous to approach 1, I do 
	app_container# bundle install
I get denied access to the fttp server with the specified ketcher-rails gem: 
![image](https://user-images.githubusercontent.com/59832692/109198591-b3cc9780-779e-11eb-8c32-e86532385bec.png)

Thanks for your time!
I sincerely hope it might help with debugging!
If the problem lies with me - please tell me the right way. 

Best regards, 
Denys

**Comment by PiTrem** [comment]
Hi @Denys-Priadko,

thx for the  report 👍  and sorry for the hassle 🙇 . I think part of the problem lies with the fact that we did not touch the docker part for a while and our docs could be better.

- About, the branches: the Master has been left behind. We develop on development (also on our production here). We ll have better versioning this year.

- about ruby version: we ve swithed to using 2.6.6: we ll have to clean that

- typo in config file
 > Adding the “ ‘ “ character.

has been fixed 👍 

- About the noVNC:  

- About the docker installation:
The Docker installation has been worked on recently. Maybe @ptrxyz could comment on that ...
What I can say is we are upping our documentation. At the moment more up to scratch docs for the docker install can be found here https://eln.chemotion.net/chemotionsaurus/docs/eln/docker_installation. ( Those docs are not mentioned anywhere .. yet)

- About the script installation:
>After linking
>#sudo ln -s /bin/mkdir /usr/bin/mkdir


interesting: I got this issue on one server only, that I kept ugrading , but not on fresh OS installation so I did not dwelve into it. So apparently it is due to some changes  between ubuntu 20.04 minor releases

>to avoid the error while nokogiri installation, at the end we get:

some gem dependencies require a bit of time and resource to compile.
After more testing, I'd recommend installing with at least 2 cpu and 3 GB memory for a server. With that the gem isntallation should run properly .  
Still the gem _bundle_  step shown in the 2/ excerpt can take around 30 mins. The last line with a warning: monitoring resource consumption will help  telling whether it has stalled or is still compiling.
